### PR TITLE
Close connections with misbehaving peers

### PR DESCRIPTION
**Pull request by kettlebell**

### CODE REVIEW

File: spectrum-network/src/peer_manager.rs [hunk 1]
```diff
@@ -524,7 +528,11 @@ impl<S: PeersState> PeerManagerNotificationsBehavior for PeerManager<S> {
         match self.state.peer(&peer_id) {
             Some(PeerInState::Connected(mut cp)) => {
                 cp.adjust_reputation(ReputationChange::NoResponse);
-                cp.disconnect();
+                if cp.get_reputation() < self.conf.misbehaving_peer_threshold {
+                    self.disconnect(peer_id, true);
```
Comment (oskin1) [inline spectrum-network/src/peer_manager.rs]: It's not the best choice to use PeerManager::disconnect() here as it emits `PeerManagerOut::Drop(peer_id)` which will be processed by NC. But in this case this is pointless as the peer is not really connected.

File: spectrum-network/src/peer_manager.rs [hunk 2]
```diff
@@ -524,7 +528,11 @@ impl<S: PeersState> PeerManagerNotificationsBehavior for PeerManager<S> {
         match self.state.peer(&peer_id) {
             Some(PeerInState::Connected(mut cp)) => {
                 cp.adjust_reputation(ReputationChange::NoResponse);
-                cp.disconnect();
+                if cp.get_reputation() < self.conf.misbehaving_peer_threshold {
```
Comment (oskin1) [inline spectrum-network/src/peer_manager.rs]: According to the docs you left under `misbehaving_peer_threshold` "misbehaving peer should be dropped", but here you also forget him. Please sync the documentation.

Also why are you checking the threshold only on dial_failure? I guess PM should check if reputation of a peer fell below the threshold each time it's adjusted (or periodically?) no matter the reason.

File: spectrum-network/src/peer_manager.rs [hunk 3]
```diff
@@ -240,6 +240,10 @@ pub struct NetworkingConfig {
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct PeerManagerConfig {
+    /// A peer with reputation below this value is classed as 'misbehaving', and its connection
+    /// should be dropped.
+    pub misbehaving_peer_threshold: Reputation,
+    /// Represents the minimum reputation a peer must have to accept its incoming connection.
     pub min_reputation: Reputation,
```
Comment (oskin1) [inline spectrum-network/src/peer_manager.rs]: I think we should keep the naming of these two settings consistent (e.g `min_drop_reputation`, `min_accept_reputation` or something like that)

Comment (kettlebell) [reply spectrum-network/src/peer_manager.rs]: Sure, will use `min_acceptable_reputation`.

File: spectrum-network/src/peer_manager.rs (hunk 2 above)
Comment (kettlebell) [reply spectrum-network/src/peer_manager.rs]: Changes to reputation now all go through `PeerManagerRequestsBehavior::on_report_peer()`, so on all adjustments. This means we'll need to use `PeerManager::disconnect(..)` whenever a peer's reputation is too low.

**Review by oskin1 — APPROVED** [review]

**Pull request merged**